src/test/java/org/zalando/nakadi/controller/EventScopeTest.java

```diff
@@ -0,0 +1,119 @@
+package org.zalando.nakadi.controller;
+
+import com.codahale.metrics.MetricRegistry;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Matchers;
+import org.mockito.Mockito;
+import org.springframework.core.MethodParameter;
+import org.springframework.http.converter.StringHttpMessageConverter;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.test.web.servlet.setup.StandaloneMockMvcBuilder;
+import org.springframework.web.bind.support.WebDataBinderFactory;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.method.support.ModelAndViewContainer;
+import org.zalando.nakadi.config.JsonConfig;
+import org.zalando.nakadi.domain.EventType;
+import org.zalando.nakadi.enrichment.Enrichment;
+import org.zalando.nakadi.exceptions.IllegalScopeException;
+import org.zalando.nakadi.metrics.EventTypeMetricRegistry;
+import org.zalando.nakadi.metrics.EventTypeMetrics;
+import org.zalando.nakadi.partitioning.PartitionResolver;
+import org.zalando.nakadi.repository.TopicRepository;
+import org.zalando.nakadi.repository.db.EventTypeCache;
+import org.zalando.nakadi.security.AuthorizedClient;
+import org.zalando.nakadi.security.Client;
+import org.zalando.nakadi.service.EventPublisher;
+import org.zalando.nakadi.utils.EventTypeTestBuilder;
+import org.zalando.nakadi.validation.EventTypeValidator;
+
+import java.util.Collections;
+import java.util.Set;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+
+public final class EventScopeTest {
+
+    private static final String EVENT_NAME = "my-event";
+    private static final String EVENT_BATCH = "[{\"payload\": \"My Event Payload\"}]";
+    private static final Set<String> SCOPE_WRITE = Collections.singleton("oauth2.scope.write");
+    private static final String URL = "/event-types/" + EVENT_NAME + "/events";
+
+    private final ObjectMapper objectMapper = new JsonConfig().jacksonObjectMapper();
+    private final EventTypeCache eventTypeCache;
+    private final StandaloneMockMvcBuilder mvcBuilder;
+
+    public EventScopeTest() throws Exception {
```

Comment:
The reason why I moved it to another class is to have not mocked EventPublisher.